### PR TITLE
Tree Graph Integration

### PR DESCRIPTION
**Pull request by nxanil**
@natabal 

Convert `Tree Graph` into the functional component using various HOCs and new library material-ui/core. Please have a look.

[Screenshot from Gyazo](https://gyazo.com/1bad277a9fde8d928f12cd8ac3420265)

[Screenshot from Gyazo](https://gyazo.com/e059216d96bc08b91096493fdbbda6b4)

Ref PR:- https://github.com/nuagenetworks/vsd-react-ui/pull/3740

### CODE REVIEW

File: GraphsV2/TreeGraph/index.js [hunk 1]
```diff
@@ -0,0 +1,598 @@
...
+            .attr('refY', 0)
+            .attr('markerWidth', 6)
+            .attr('markerHeight', 6)
+            .attr('orient', 'auto')
+            .attr('fill', linksSettings.stroke.selectedColor)
+            .append('path')
+            .attr('d', 'M10,-5L0,0L10,5');
+    }
+
+    const diagonal = d => {
```
Comment (natabal) [inline GraphsV2/TreeGraph/index.js]: move out of the component

File: GraphsV2/TreeGraph/index.js [hunk 2]
```diff
@@ -0,0 +1,598 @@
...
+            .attr('refY', 0)
+            .attr('markerWidth', 6)
+            .attr('markerHeight', 6)
+            .attr('orient', 'auto')
+            .attr('fill', linksSettings.stroke.defaultColor)
+            .append('path')
+            .attr('d', 'M10,-5L0,0L10,5');
+    }
+
+    const coloredArrow = (svg, linksSettings) => {
```
Comment (natabal) [inline GraphsV2/TreeGraph/index.js]: this does not depend on any of the component state, let's move it out of component.

File: GraphsV2/TreeGraph/index.js [hunk 3]
```diff
@@ -0,0 +1,598 @@
...
+            .attr('marker-start', d => d.data.clicked ? 'url(#colored-arrow)' : 'url(#normal-arrow)');
+
+        // Remove any exiting links
+        link.exit().transition()
+            .duration(duration)
+            .attr('d', d => diagonal(d))
+            .remove();
+    }
+
+    const normalArrow = (svg, linksSettings) => {
```
Comment (natabal) [inline GraphsV2/TreeGraph/index.js]: move it out of component

File: GraphsV2/TreeGraph/index.js [hunk 4]
```diff
@@ -0,0 +1,598 @@
...
+        node.exit().transition()
+            .duration(duration)
+            .attr('transform', () => `translate(${source.y}, ${source.x})`)
+            .remove();
+
+        nodeUpdate.select('rect')
+            .style('fill', d => d.data.clicked ? rectNode.selectedBackground : rectNode.defaultBackground);
+    }
+
+    const renderRectNode = d => {
```
Comment (natabal) [inline GraphsV2/TreeGraph/index.js]: this could be move out of component.

**Comment by natabal** [comment]
@nxanil Last meeting we had I asked for the paths on each node to be changed! I don't see those changes. Can you please update this PR with requested changes and please add screenshots.

**Comment by natabal** [comment]
@nxanil Did you get a chance to import the changes you have made to master TreeGraph? If yes. Let's merge this PR.

**Comment by nxanil** [comment]
@natabal 
I have made the changes.

[Screenshot from Gyazo](https://gyazo.com/51600236978ec7971fb9a784ad105478)

**Pull request merged**